tests/integration/cases/django_with_dep_resolution_virtual_env_as_input/test.yaml

```diff
@@ -36,6 +36,29 @@ steps:
     kind: policy_report
     result: output/policy_report.json
     expected: policy_report.json
+- name: Clean up the database.
+  kind: shell
+  options:
+    cmd: rm -f output/macaron.db
```

Comment:
I wonder why we need to clean up the database here, is it because the policy we use after this step will fail for the component that we analyze in the previous steps ? :thinking:

Reply:
No, the policy doesn't fail if we don't clean up the database, but I prefer to make sure the results from the previous step does not affect this step.

Reply:
Hmm if we are expecting it to not fail, it failing unexpectedly would be a good sign for a potential issue/bug. Cleaning it here would prevent us from noticing such issue happening, especially if 3 analyze command are run sequentially similarly to the tutorial.

Reply:
The three `analyze` commands in the tutorial are not expected to run sequentially. 

On the contrary, the risk with not removing the cached `macaron.db` between steps is that a test might pass only because the previous step has populated facts accidentally. So, if `analyze` command has run again and a check is unexpectedly failed, depending on the policy, if the target PURL matches the previous run of `analyze` command, we might miss the error.

Reply:
Hmm in this case, do you think it's worthwhile separating them into another test directory :thinking: ?

Reply:
I don't see any issue with keeping the tests together.

>In additions, if three commands in the tutorial are not expected to run sequentially, should we instruct the user to delete the database too?

Different use cases in this tutorial can run in any order and that doesn't matter. We definitely shouldn't assume they run the examples in order. Regardless of the order, users need to be careful with bugs in their policies if they use the same cached outputs between runs in general and that's independent of this tutorial.